File: back/internal/domain/food.go
package domain

import "time"

type Food struct {
	CaloriesPer100 int
	Carbs          int
	CreatedByAdmin bool
	Fat            int
	ID             int
	Name           string
	Protein        int
	UserID         int
}

type FoodInMeal struct {
	Calories       int
	CaloriesPer100 int
	Name           string
	Weight         int
	MealId         int
}

type Meal struct {
	Calories int
	Date     time.Time
	Id       int
	Name     string
	Foods    []FoodInMeal
	UserId   int
}

File: front/src/types/food.ts
export type FoodResponse = {
  name: string;
  caloriesPer100: number;
  carbs: number;
  fat: number;
  id: number;
  protein: number;
};

export type MealResponse = {
  calories: number;
  date: Date;
  id: number;
  name: string;
  foods: FoodInMealResponse[];
};

type FoodInMealResponse = {
  calories: number;
  name: string;
  weight: number;
};
